pippy/ManualPipelineStage.py

```diff
@@ -306,9 +307,15 @@ def __init__(
         self.inputs_grad: List[torch.tensor] = []
         self.outputs_grad: List[torch.tensor] = []
 
-        # TODO: Calculating stage index from rank is not ideal, e.g. won't match in Interleaved 1F1B case.
-        self.prev_stage = (self.rank - 1) % self.world_size
-        self.next_stage = (self.rank + 1) % self.world_size
+        def stage_global_rank(peer_rank):
+            return (
+                peer_rank
+                if self.group is None
+                else dist.get_global_rank(self.group, peer_rank)
+            )
+
+        self.prev_stage = stage_global_rank((self.rank - 1) % self.world_size)
+        self.next_stage = stage_global_rank((self.rank + 1) % self.world_size)
```

Comment:
nit: how will we deal with the Interleaved case?

Reply:
in manual pipeline stages there is no support for skip connections (multiple prev and next ranks) so this works